### PR TITLE
Appcues v3

### PR DESCRIPTION
**Pull request by gbova**
There's been some concern that the template apps for Appcues take too long to load, and delay the confirmation email (which is bad for a new user's experience). This PR loads the three template apps in parallel and sends the confirmation email to the user after all three have finished loading.

I'm still not 100% confident this will fix the issue we discussed over email, where it takes 1.5 - 7 minutes for the confirmation email to show up in your inbox. We keep the user on the "Creating Account..." loading page until we've done everything on the backend that we need to:
creating_account

We only show the "Success! An email was just sent to ..." page after the apps have been loaded and a call to `send_domain_registration_email` has been made:
success

Seeing as this success page loads for me in seconds, but I don't receive the confirmation email for a while, there may still be an issue with `send_domain_registration_email` or the queue it's on (email_queue).

Either way, I'm hopeful that loading these apps in parallel can only help the situation

Jira ticket: https://trello.com/c/BcDom7IC/114-performance-improvements-activation-email-delay?menu=filter&filter=member:gabriellabova1

Original Appcues specs: https://docs.google.com/document/d/17ml1B5evU3Z45Y2a5q4Nrrli-nWVhlL-PAkvrzMlFt0/edit# 
Original Appcues PR: https://github.com/dimagi/commcare-hq/pull/22487/files

### CODE REVIEW

File: corehq/apps/registration/utils.py [hunk 1]
```diff
@@ -138,19 +134,20 @@ def request_new_domain(request, form, is_new_user=True):
...
+                for slug in APPCUES_APP_SLUGS
+            ]
+            callback = send_domain_registration_email.si(
+                request.user.email,
+                dom_req.domain,
+                dom_req.activation_guid,
+                request.user.get_full_name(),
+                request.user.first_name
+            )
+            chord(header)(callback)
```
Comment (dannyroberts) [inline corehq/apps/registration/utils.py]: Such funny syntax :) Does this work okay? In some of the examples it looked like the list of results of the header tasks was passed in as an argument to the callback. Just to check that this works without having to change the signature of the `send_domain_registration_email` function.

**Review by dannyroberts — APPROVED** [review]

File: corehq/apps/registration/utils.py (hunk 1 above)
Comment (gbova) [reply corehq/apps/registration/utils.py]: I do see what you mean about the callback taking the results of the functions in the header, but the apps still load correctly and before the email is sent. What does your gut tell you could be going wrong?

Comment (dannyroberts) [reply corehq/apps/registration/utils.py]: Yeah if you locally make sure `CELERY_ALWAYS_EAGER` is `False`, stop celery, start celery, and then test this through the UI then it's probably fine.

The thing I was wondering about was whether the callback task (`send_domain_registration_email`) needed to take an extra argument (the list of results of the headers) for this to work. But if it works then I guess not.

**Comment by proteusvacuum** [comment]
This seems like a good solution, but have you considered sending the email synchronously before you kick off these tasks so that the user isn't blocked from signing up (even if celery is backed up)? 

If the user confirms before the task is complete, you could provide some feedback that the test apps are still generating with a banner at the top with a spinner that says `Creating sample applications`, that could go away once that task is complete.

I know that when I'm signing up for a website I expect the email to come immediately, otherwise I'll go do something else and probably won't go back.

**Comment by gbova** [comment]
It's difficult to completely eliminate frustration in the signup flow. When I first picked this up in December, Marshall, Jenny, and I had a conversation about how this workflow should look. Marshall argued that we shouldn't give the new user access to CommCare (i.e. we shouldn't send them their confirmation email) until after the apps had been loaded because. The way we saw it, we had three options:
1. Send the confirmation email before the template apps have finished loading, and, when the user goes to confirm their account, show them a loading page and don't allow them to enter CommCare. Marshall argued that users tend to get annoyed when they are asked to log in for the first time and then are blocked from the site. He thought that if a user saw a loading page after confirming their account, they'd just quit out and never come back.
2. Send the confirmation email before the template apps have finished loading, and, when the user goes to confirm their account, send them to the dashboard (where users normally land) and show them the regular login flow; then, the next time they log in and as long as the apps are ready, throw them into the new Appcues flow. We decided against this workflow because allowing first-time users to enter CommCare before seeing the Appcues flow renders the new feature pretty useless, and users might get confused as to why they are seeing something entirely new. (Also this option might be impossible because, if I remember correctly, Appcues relies on users hitting the site for the first time to work).
3. Wait to send the confirmation email until after the apps have been loaded into the user's domain. This way, we are guaranteed that everything for Appcues will be ready when the user logs in for the first time. We knew that there was a chance loading the apps could take a long time. Marshall reduced the size of the apps as much as he could to reduce the load time. Parallel-izing the way we load the apps (i.e. this PR) is the second step we've taken to reduce the lag. Plus, in my experience, loading the apps doesn't take long at all; receiving the confirmation email in your inbox is what takes a long time

**Comment by proteusvacuum** [comment]
Got it, thanks for explanation. I was missing the part where people were given a different view of HQ the first time they log in.

Since this seems like a pretty mission-critical part of the sign-up flow, can we make a separate queue for signup emails and stick that on an underutilized machine? That way those emails aren't blocked by other things in the email queue?

**Comment by gbova** [comment]
I like that idea, but I didn't realize I could make a new one. Do we ever try to limit the number of queues we have?

**Comment by proteusvacuum** [comment]
I don't think we try to limit the number of queues we have currently.

I think to add a new one you just add the name here: https://github.com/dimagi/commcare-cloud/blob/7d0a2e092519a33f738f39280d69a6428ecd8b02/environments/production/app-processes.yml#L14 (and for the other envs)

You could also think about either bumping the concurrency on the email queue or also adding it to a different machine as well so that there are more machines working on clearing that queue.

**Pull request merged**